src/config.ts

```diff
@@ -509,6 +512,32 @@ export class KubeConfig implements SecurityAuthentication {
         return this.getContextObject(this.currentContext);
     }
 
+    private createAgent(
+        cluster: Cluster | null,
+        agentOptions: https.AgentOptions,
+    ): https.Agent | SocksProxyAgent | HttpProxyAgent | HttpsProxyAgent {
+        let agent: https.Agent | SocksProxyAgent | HttpProxyAgent | HttpsProxyAgent;
+
+        if (cluster && cluster.proxyUrl) {
+            if (cluster.proxyUrl.startsWith('socks')) {
+                agent = new SocksProxyAgent(cluster.proxyUrl, agentOptions);
+            } else if (cluster.server.startsWith('https')) {
```

Comment:
Using either HttpsProxyAgent or HttpProxyAgent doesn't seem to depend on the proxy protocol but on the server protocol it seems

seem the first table of their documentation: https://github.com/delvedor/hpagent?tab=readme-ov-file#usage

Reply:
In case of HTTP and HTTPS proxy, you are right, and it's implemented: `(cluster.server.startsWith('https'))`. In case of socks proxy, we check proxy URL protocol in order to recognize its socks. That is why socks condition is before HTTPS and HTTP.

Reply:
Oh I'm very sorry, I've read the code too fast!